libraries/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -238,20 +241,30 @@ static IotMqttError_t _subscriptionCommon( IotMqttOperationType_t operation,
 
 /*-----------------------------------------------------------*/
 
+/**
+ * @brief Uninitialized value for @ref _initCalled.
+ */
+#define MQTT_LIBRARY_UNINITIALIZED    ( ( uint32_t ) 0 )
+
+/**
+ * @brief Initialized value for @ref _initCalled.
+ */
+#define MQTT_LIBRARY_INITIALIZED      ( ( uint32_t ) 1 )
+
 /**
  * @brief Tracks whether @ref mqtt_function_init has been called.
  *
  * API functions will fail if @ref mqtt_function_init was not called.
  */
-static uint32_t _initCalled = 0;
+static volatile uint32_t _initCalled = MQTT_LIBRARY_UNINITIALIZED;
```

Comment:
Does this need to be declared volatile?  Probably not, it's not intended to be a variable shared between threads.

Reply:
It has to be `volatile` in order to [avoid a compiler warning.](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/v4_beta/ports/common/include/atomic/iot_atomic_generic.h#L98)